# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1503611491 25200
#      Thu Aug 24 14:51:31 2017 -0700
# Node ID fd6b1339689f7a72ce4860b487aa32ff4562558e
# Parent  e54e13ee99ce1a29caf084ce43671be2553dfde1
build: set FLTK_LIBS properly by splitting FLTK_LDFLAGS

* configure.ac: Split the output of fltk-config --ldflags into FLTK_LDFLAGS
and FLTK_LIBS by value.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1933,17 +1933,28 @@ if test $check_fltk = yes; then
     fltk_cflags=`$FLTK_CONFIG $fltkconf_args --use-gl --cflags`
     for fltk_option in $fltk_cflags; do
       case $fltk_option in
         -I*)
           FLTK_CPPFLAGS="$FLTK_CPPFLAGS $fltk_option"
         ;;
       esac
     done
-    FLTK_LDFLAGS=`$FLTK_CONFIG $fltkconf_args --use-gl --ldflags`
+    ## Split -L and -l options into FLTK_LDFLAGS and FLTK_LIBS
+    fltk_ldflags=`$FLTK_CONFIG $fltkconf_args --use-gl --ldflags`
+    for fltk_option in $fltk_ldflags; do
+      case $fltk_option in
+        -l*)
+          FLTK_LIBS="$FLTK_LIBS $fltk_option"
+        ;;
+        *)
+          FLTK_LDFLAGS="$FLTK_LDFLAGS $fltk_option"
+        ;;
+      esac
+    done
 
     case $host_os in
       mingw*)
         FLTK_LDFLAGS=`echo $FLTK_LDFLAGS | $SED -e 's/-mwindows//g'`
       ;;
     esac
     AC_LANG_PUSH(C++)
     AC_CACHE_CHECK([for OpenGL support in FLTK],
